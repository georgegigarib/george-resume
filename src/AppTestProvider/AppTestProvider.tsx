import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
