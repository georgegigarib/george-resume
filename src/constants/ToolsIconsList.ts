import JsIcon from '@/assets/icons/tools/Javascript.svg'

export type ToolInfo = {
  name: string
  icon: IconName
  color: string
  bgColor: string
  darkBgColor?: string
  borderColor: string
  darkBorderColor?: string
  darkColor?: string
  svg?: string
  custom?: boolean
}

export type ToolIconProps = {
  tool: IconName
  size?: '5x' | '4x' | '3x' | '2x' | 'xl' | 'lg' | 'md' | 'sm' | 'xs'
}

export enum IconName {
  React = 'react',
  ReactNative = 'reactnative',
  TailwindCSS = 'tailwindcss',
  JavaScript = 'js',
  TypeScript = 'ts-circle',
  VueJs = 'vuejs',
  NextJS = 'nextjs',
  NodeJS = 'nodejs',
  HTML = 'html',
  CSS = 'css',
  Express = 'express',
  Spotify = 'spotify',
  Git = 'git',
  Redis = 'redis',
  Bun = 'bun',
  Docker = 'docker',
  Pinia = 'pinia',
  Sass = 'sass',
  Bootstrap = 'bootstrap',
  Angular = 'angular',
  C = 'c',
  CSharp = 'csharp',
  Django = 'django',
  FramerMotion = 'framer-motion',
  Go = 'go',
  Graphql = 'graphql',
  Javalanguage = 'java',
  Laravel = 'laravel',
  Mui = 'mui',
  Jest = 'jest',
  Vitest = 'vitest',
  Cypress = 'cypress',
  MySQL = 'mysql',
  MongoDb = 'mongodb',
  Php = 'php',
  Postgresql = 'postgresql',
  Python = 'python',
  Redux = 'redux',
  Zustand = 'zustand',
  Svelte = 'svelte',
}

export const toolsIconsList: ToolInfo[] = [
  {
    name: 'React',
    icon: IconName.React,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'React',
    icon: IconName.ReactNative,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Tailwind CSS',
    icon: IconName.TailwindCSS,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#38BDF8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'JavaScript',
    icon: IconName.JavaScript,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
    custom: true,
    svg: JsIcon,
  },
  {
    name: 'Jest',
    icon: IconName.Jest,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MySQL',
    icon: IconName.MySQL,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#38BDF8',
  },
  {
    name: 'Vitest',
    icon: IconName.Vitest,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
  },
  {
    name: 'Cypress',
    icon: IconName.Cypress,
    color: '#000000FF',
    bgColor: '#B4B4B4FF',
    darkBgColor: '#BABABAFF',
    borderColor: '#C7B008FF',
  },
  {
    name: 'TypeScript',
    icon: IconName.TypeScript,
    color: '#3178C6',
    bgColor: '#FAFAFAFF',
    borderColor: '#3178C6',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Next.js',
    icon: IconName.NextJS,
    color: '#0000000',
    bgColor: '#FAFAFAFF',
    borderColor: '#555555FF',
    darkBgColor: '#3E3E3EFF',
    darkColor: '#FFFFFF',
    darkBorderColor: '#BFBFBFFF',
  },
  {
    name: 'Node.js',
    icon: IconName.NodeJS,
    color: '#539E43',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#539E43',
  },
  {
    name: 'HTML',
    icon: IconName.HTML,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'CSS',
    icon: IconName.CSS,
    color: '#264DE4',
    bgColor: '#FAFAFAFF',
    borderColor: '#264DE4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Express',
    icon: IconName.Express,
    color: '#333333',
    bgColor: '#FAFAFAFF',
    borderColor: '#CCCCCC',
  },
  {
    name: 'Spotify',
    icon: IconName.Spotify,
    color: '#1DB954',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#1DB954',
  },
  { name: 'Git', icon: IconName.Git, color: '#F05032', bgColor: '#F5F5F5', borderColor: '#F05032' },
  {
    name: 'Redis',
    icon: IconName.Redis,
    color: '#D82C20',
    bgColor: '#FAFAFAFF',
    borderColor: '#D82C20',
  },
  {
    name: 'Bun',
    icon: IconName.Bun,
    darkColor: '#ffffff',
    color: '#333333',
    bgColor: '#FAFAFAFF',
    darkBorderColor: '#bababa',
    borderColor: '#686767FF',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Docker',
    icon: IconName.Docker,
    color: '#2496ED',
    bgColor: '#FAFAFAFF',
    borderColor: '#2496ED',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Pinia',
    icon: IconName.Pinia,
    color: '#F0C723FF',
    darkBgColor: '#2C2C2C',
    bgColor: '#FAFAFAFF',
    borderColor: '#FFD939',
  },
  {
    name: 'Sass',
    icon: IconName.Sass,
    color: '#CC6699',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#CC6699',
  },
  {
    name: 'Bootstrap',
    icon: IconName.Bootstrap,
    color: '#A877F3FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#7952B3',
  },
  {
    name: 'Angular',
    icon: IconName.Angular,
    color: '#DD0031',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#DD0031',
  },
  {
    name: 'C',
    icon: IconName.C,
    color: '#00599C',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#00599C',
  },
  {
    name: 'C#',
    icon: IconName.CSharp,
    color: '#239120',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#239120',
  },
  {
    name: 'Django',
    icon: IconName.Django,
    color: '#092E20',
    bgColor: '#FAFAFAFF',
    borderColor: '#092E20',
  },
  {
    name: 'Framer Motion',
    icon: IconName.FramerMotion,
    color: '#EC5990',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#EC5990',
  },
  {
    name: 'Go',
    icon: IconName.Go,
    color: '#00ADD8',
    bgColor: '#FAFAFAFF',
    borderColor: '#00ADD8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'GraphQL',
    icon: IconName.Graphql,
    color: '#E10098',
    bgColor: '#FAFAFAFF',
    borderColor: '#E10098',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'java',
    icon: IconName.Javalanguage,
    color: '#007396',
    bgColor: '#FAFAFAFF',
    borderColor: '#007396',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Laravel',
    icon: IconName.Laravel,
    color: '#FF2D20',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF2D20',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Material UI',
    icon: IconName.Mui,
    color: '#0081CB',
    bgColor: '#FAFAFAFF',
    borderColor: '#0081CB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MongoDB',
    icon: IconName.MongoDb,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Vue Js',
    icon: IconName.VueJs,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PHP',
    icon: IconName.Php,
    color: '#959AE1FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#777BB4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PostgreSQL',
    icon: IconName.Postgresql,
    color: '#336791',
    bgColor: '#FAFAFAFF',
    borderColor: '#336791',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Python',
    icon: IconName.Python,
    color: '#3776AB',
    bgColor: '#FAFAFAFF',
    borderColor: '#3776AB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Redux',
    icon: IconName.Redux,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Zustand',
    icon: IconName.Zustand,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Svelte',
    icon: IconName.Svelte,
    color: '#FF3E00',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF3E00',
    darkBgColor: '#2C2C2C',
  },
]
