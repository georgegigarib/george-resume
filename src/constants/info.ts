export const WORK_EMAIL = 'georgegil.work@gmail.com'
export const SOCIAL_LINKS = {
  github: 'https://github.com/georgegigarib',
  linkedin: 'https://www.linkedin.com/in/georgegg/',
  instagram: 'URL_ADDRESS.com/GeorgeGil',
}
