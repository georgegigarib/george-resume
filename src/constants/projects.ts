import image1 from '@/assets/images/projects/movie-app.gif'
import image2 from '@/assets/images/projects/mobile-2.jpeg'
import image3 from '@/assets/images/projects/mobile-1.jpg'
import image4 from '@/assets/images/projects/desktop-app-1.png'
import image5 from '@/assets/images/projects/desktop-app-2.png'
import image6 from '@/assets/images/projects/desktop-app-3.png'
import { IconName } from './ToolsIconsList'

export type ProjectInfoType = {
  src: string
  title: string
  description: string
  url: string
  tools: IconName[]
}

export const mobileProjects: ProjectInfoType[] = [
  {
    src: image1,
    title: 'Movie App',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto y debo alargarlo mas y mas y mas',
    url: 'https://movie-app.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.JavaScript]
  },
  {
    src: image2,
    title: 'Proyecto Mobile 1',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project1.example.com',
    tools: [IconName.HTML, IconName.CSS, IconName.JavaScript]
  },
  {
    src: image3,
    title: 'Proyecto Mobile 2',
    description:
      ' Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto, Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project2.example.com',
    tools: [IconName.React, IconName.TypeScript, IconName.JavaScript]
  },
  {
    src: image1,
    title: 'Proyecto Mobile 2',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto, Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project2.example.com',
    tools: [IconName.React, IconName.TypeScript]
  },
  {
    src: image2,
    title: 'Proyecto Mobile 1',
    description: 'Proyecto con animaciones personalizadas.',
    url: 'https://project1.example.com',
    tools: [IconName.HTML, IconName.CSS]
  }
]

export const desktopProjects: ProjectInfoType[] = [
  {
    src: image6,
    title: 'Proyecto Desktop 1',
    description: 'Dashboard avanzado.',
    url: 'https://desktop1.example.com',
    tools: [
      IconName.React,
      IconName.TailwindCSS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git
    ]
  },
  {
    src: image5,
    title: 'Proyecto Desktop 2',
    description:
      'Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git
    ]
  },
  {
    src: image4,
    title: 'Proyecto Desktop 3',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  },
  {
    src: image5,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.NextJS, IconName.NodeJS, IconName.TailwindCSS, IconName.Redis, IconName.Git]
  },
  {
    src: image6,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  },
  {
    src: image6,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  }
]
