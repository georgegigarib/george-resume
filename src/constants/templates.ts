import { IconName } from '@/constants/ToolsIconsList'

export interface Template {
  tools: IconName[]
  url: string
  hint: string
  bgColor: string
  description: string
}

export const devTemplates: Template[] = [
  {
    tools: [IconName.TypeScript, IconName.React, IconName.TailwindCSS, IconName.Bun, IconName.CSS],
    url: 'https://github.com/example/fast-render-template',
    hint: 'dev.templates.templates.template1.hint',
    bgColor: '#FF4500',
    description: 'dev.templates.templates.template1.description',
  },
  {
    tools: [IconName.TypeScript, IconName.NextJS, IconName.TailwindCSS, IconName.Docker, IconName.React],
    url: 'https://github.com/example/next-ssr-template',
    hint: 'dev.templates.templates.template2.hint',
    bgColor: '#FF1493',
    description: 'dev.templates.templates.template2.description',
  },
  {
    tools: [IconName.TypeScript, IconName.VueJs, IconName.TailwindCSS, IconName.Pinia, IconName.Docker, IconName.HTML],
    url: 'https://github.com/example/vue-pinia-template',
    hint: 'dev.templates.templates.template3.hint',
    bgColor: '#FFA500',
    description: 'dev.templates.templates.template3.description',
  },
  {
    tools: [IconName.JavaScript, IconName.Angular, IconName.Sass, IconName.Docker, IconName.NodeJS],
    url: 'https://github.com/example/angular-enterprise-template',
    hint: 'dev.templates.templates.template4.hint',
    bgColor: '#A26B5AFF',
    description: 'dev.templates.templates.template4.description',
  },
  {
    tools: [IconName.TypeScript, IconName.React, IconName.Mui, IconName.Graphql, IconName.Django],
    url: 'https://github.com/example/react-graphql-template',
    hint: 'dev.templates.templates.template5.hint',
    bgColor: '#877F95FF',
    description: 'dev.templates.templates.template5.description',
  },
  {
    tools: [IconName.TypeScript, IconName.Svelte, IconName.TailwindCSS, IconName.Python, IconName.Postgresql],
    url: 'https://github.com/example/svelte-minimal-template',
    hint: 'dev.templates.templates.template6.hint',
    bgColor: '#687969FF',
    description: 'dev.templates.templates.template6.description',
  },
  {
    tools: [IconName.JavaScript, IconName.React, IconName.Redux, IconName.Bootstrap, IconName.Php],
    url: 'https://github.com/example/ecommerce-template',
    hint: 'dev.templates.templates.template7.hint',
    bgColor: '#908C63FF',
    description: 'dev.templates.templates.template7.description',
  },
  {
    tools: [IconName.TypeScript, IconName.React, IconName.Graphql, IconName.MongoDb, IconName.Docker],
    url: 'https://github.com/example/data-intensive-template',
    hint: 'dev.templates.templates.template8.hint',
    bgColor: '#96659FFF',
    description: 'dev.templates.templates.template8.description',
  },
]
