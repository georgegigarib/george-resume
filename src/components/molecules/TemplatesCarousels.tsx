import clsx from 'clsx'
import { motion } from 'framer-motion'
import styles from '@/styles/TemplatesCarousels.module.css'
import ToolIcon from '../atoms/ToolIcon'
import { devTemplates } from '@/constants/templates'

export default function TemplatesCarousels() {
  const iconGroupsWithHints = devTemplates.slice(0, 3).map((project) => ({
    tools: project.tools.slice(0, 5),
    hint: project.hint,
    bgColor: project.bgColor
  }))

  return (
    <div className="col-span-1 self-center w-full rounded-xl md:flex flex-col row-span-2 hidden md:h-[400px]">
      <div className="grid grid-cols-1 gap-y-10 mt-4 px-2">
        {iconGroupsWithHints.map((groupData, groupIndex) => (
          <div key={groupIndex}>
            <div className={styles.container}>
              <div
                className={clsx(
                  styles.base,
                  {
                    [styles.reverse]: groupIndex % 2 !== 0
                  },
                  groupIndex % 2 !== 0 ? 'left-14' : 'right-14'
                )}
              >
                {groupData.tools.map((tool, index) => (
                  <div
                    key={index}
                    className={clsx(styles.icon, {
                      [styles.reverse]: groupIndex % 2 !== 0
                    })}
                  >
                    <ToolIcon tool={tool} size="2x" />
                  </div>
                ))}
              </div>
            </div>
            <motion.div
              className={clsx('relative -top-20', groupIndex % 2! === 0 ? 'left-32' : ' left-0')}
              whileHover={{ scale: 1.03 }}
            >
              <div
                className="absolute flex items-center text-center flex-wrap max-w-24 h-20 p-1 rounded-xl"
                style={{ backgroundColor: groupData.bgColor }}
              >
                <p className="font-bold capitalize text-sm">{groupData.hint}</p>
              </div>
            </motion.div>
          </div>
        ))}
      </div>
    </div>
  )
}
