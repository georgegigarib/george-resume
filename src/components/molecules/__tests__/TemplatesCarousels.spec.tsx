import { render, screen } from '@testing-library/react'
import { devTemplates } from '@/constants/templates'
import { describe, it, expect, vi } from 'vitest'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'
import TemplatesCarousels from '@/components/molecules/TemplatesCarousels'

// Mock the devTemplates to control the test environment
vi.mock('@/constants/templates', () => ({
  devTemplates: [
    {
      tools: ['React', 'TypeScript', 'Jest'],
      hint: 'template1',
      bgColor: '#fff',
      url: 'http://example.com/template1',
    },
    {
      tools: ['Node.js', 'Express', 'MongoDB'],
      hint: 'template2',
      bgColor: '#f0f0f0',
      url: 'http://example.com/template2',
    },
    {
      tools: ['Python', 'Django', 'PostgreSQL'],
      hint: 'template3',
      bgColor: '#e0e0e0',
      url: 'http://example.com/template3',
    },
  ],
}))

describe('TemplatesCarousels Component', () => {
  const renderComponent = () =>
    render(
      <AppTestProvider>
        <TemplatesCarousels />
      </AppTestProvider>
    )

  it('renders correctly with provided templates', () => {
    const { asFragment } = renderComponent()

    devTemplates.forEach((template) => {
      template.tools.forEach((tool) => {
        expect(screen.getByLabelText(tool)).toBeInTheDocument()
      })
    })

    expect(asFragment()).toMatchSnapshot()
  })
})
