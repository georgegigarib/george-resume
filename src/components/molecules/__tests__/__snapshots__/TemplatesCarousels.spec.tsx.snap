// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TemplatesCarousels Component > renders correctly with provided templates 1`] = `
<DocumentFragment>
  <div
    class="col-span-1 self-center w-full rounded-xl md:flex flex-col row-span-2 hidden md:h-[420px]"
  >
    <div
      class="grid grid-cols-1 gap-y-10 mt-4 px-2"
    >
      <div>
        <div
          class="_container_e676c2"
        >
          <div
            class="_base_e676c2 right-14"
          >
            <div
              aria-label="React"
              class="_icon_e676c2"
              data-mui-internal-clone-element="true"
            />
            <div
              aria-label="TypeScript"
              class="_icon_e676c2"
              data-mui-internal-clone-element="true"
            />
            <div
              aria-label="Jest"
              class="_icon_e676c2"
              data-mui-internal-clone-element="true"
            />
          </div>
        </div>
        <div
          class="relative -top-20 group left-32"
        >
          <a
            href="http://example.com/template1"
          >
            <div
              class="absolute flex items-center text-center flex-wrap max-w-24 h-20 p-1 rounded-xl"
              style="background-color: rgb(255, 255, 255);"
            >
              <div
                aria-label="See template"
                class="absolute -top-1 right-0 opacity-0 group-hover:opacity-100 transition-opacity hover:bg-white/25 bg-tra px-1 mt-1 rounded-full"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit -top-[2px] relative css-1lzspe0-MuiSvgIcon-root"
                  data-testid="ArrowOutwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 6v2h8.59L5 17.59 6.41 19 16 9.41V18h2V6z"
                  />
                </svg>
              </div>
              <p
                class="font-bold capitalize text-sm text-black dark:text-white"
              />
            </div>
          </a>
        </div>
      </div>
      <div>
        <div
          class="_container_e676c2"
        >
          <div
            class="_base_e676c2 _reverse_e676c2 left-14"
          >
            <div
              aria-label="Node.js"
              class="_icon_e676c2 _reverse_e676c2"
              data-mui-internal-clone-element="true"
            />
            <div
              aria-label="Express"
              class="_icon_e676c2 _reverse_e676c2"
              data-mui-internal-clone-element="true"
            />
            <div
              aria-label="MongoDB"
              class="_icon_e676c2 _reverse_e676c2"
              data-mui-internal-clone-element="true"
            />
          </div>
        </div>
        <div
          class="relative -top-20 group  left-0"
        >
          <a
            href="http://example.com/template2"
          >
            <div
              class="absolute flex items-center text-center flex-wrap max-w-24 h-20 p-1 rounded-xl"
              style="background-color: rgb(240, 240, 240);"
            >
              <div
                aria-label="See template"
                class="absolute -top-1 right-0 opacity-0 group-hover:opacity-100 transition-opacity hover:bg-white/25 bg-tra px-1 mt-1 rounded-full"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit -top-[2px] relative css-1lzspe0-MuiSvgIcon-root"
                  data-testid="ArrowOutwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 6v2h8.59L5 17.59 6.41 19 16 9.41V18h2V6z"
                  />
                </svg>
              </div>
              <p
                class="font-bold capitalize text-sm text-black dark:text-white"
              />
            </div>
          </a>
        </div>
      </div>
      <div>
        <div
          class="_container_e676c2"
        >
          <div
            class="_base_e676c2 right-14"
          >
            <div
              aria-label="Python"
              class="_icon_e676c2"
              data-mui-internal-clone-element="true"
            />
            <div
              aria-label="Django"
              class="_icon_e676c2"
              data-mui-internal-clone-element="true"
            />
            <div
              aria-label="PostgreSQL"
              class="_icon_e676c2"
              data-mui-internal-clone-element="true"
            />
          </div>
        </div>
        <div
          class="relative -top-20 group left-32"
        >
          <a
            href="http://example.com/template3"
          >
            <div
              class="absolute flex items-center text-center flex-wrap max-w-24 h-20 p-1 rounded-xl"
              style="background-color: rgb(224, 224, 224);"
            >
              <div
                aria-label="See template"
                class="absolute -top-1 right-0 opacity-0 group-hover:opacity-100 transition-opacity hover:bg-white/25 bg-tra px-1 mt-1 rounded-full"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit -top-[2px] relative css-1lzspe0-MuiSvgIcon-root"
                  data-testid="ArrowOutwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 6v2h8.59L5 17.59 6.41 19 16 9.41V18h2V6z"
                  />
                </svg>
              </div>
              <p
                class="font-bold capitalize text-sm text-black dark:text-white"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
